onnxruntime/core/providers/cuda/cuda_graph.cc

```diff
@@ -29,40 +60,48 @@ void CUDAGraph::CaptureBegin() {
   CUDA_CALL_THROW(cudaStreamBeginCapture(stream_, cudaStreamCaptureModeGlobal));
 }
 
-void CUDAGraph::CaptureEnd() {
-  CUDA_CALL_THROW(cudaStreamEndCapture(stream_, &graph_));
-  if (graph_ == NULL) {
+void CUDAGraphManager::CaptureEnd() {
+  ORT_ENFORCE(IsGraphCaptureAllowedOnRun());
```

Comment:
Nit: Mildly confusing that we need to enforce is graph capture is allowed for this Run() while ending graph capture. Since, we do the enforce at `CaptureBegin()`, isn't that enough ?